f600233d1b154d507b782f84604bcb12/update/sys_script_include_b29002b097a211106fa8b4b3f153afc4.xml

```diff
@@ -5,1190 +5,1527 @@
+                packageSysId :
+                this.acquisitionPackage.sys_id;
+
+            var payloadString = this.getPayload(packageId);
+            var payload = JSON.parse(payloadString);
+
+            // transform IGCE payload
+            var uniqueClins = payload.templatePayload.periodsEstimate
+                .flatMap(function(estimate) {
+                    return estimate.periodLineItems.map(function(lineItem) {
+                        return lineItem.idiqClin;
+                    });
+                })
+                .filter(function(lineItem, index, self) {
+                    return self.indexOf(lineItem) === index;
+                });
+
+            var uniquePeriods = [];
+            var data = uniqueClins.map(function(clin) {
+                var entry = {
+                    "CLIN Type & Classification": clin,
+                };
+                payload.templatePayload.periodsEstimate.forEach(function(
+                    estimate
+                ) {
+                    var key;
+                    if (estimate.period.periodType === "BASE") {
+                        key = "Base Period";
+                    } else if (estimate.period.periodType === "OPTION") {
+                        key = "Option " + (estimate.period.optionOrder - 1);
+                    }
+                    if (uniquePeriods.indexOf(key) === -1) {
+                        uniquePeriods.push(key);
+                    }
+                    var value = estimate.periodLineItems
+                        .filter(function(lineItem) {
+                            return lineItem.idiqClin === clin;
+                        })
+                        .reduce(function(result, lineItem) {
+                            return result + lineItem.unitPrice * lineItem.quantity;
+                        }, 0);
+                    entry[key] = value;
+                });
+
+                entry.Total = sum(entry);
+                return entry;
+            });
+
+            // Calculation of subtotals for each period
+            var subtotals = uniquePeriods.reduce(function(periodResult, period) {
+                periodResult[period] = data
+                    .filter(function(entry) {
+                        return entry[period];
+                    })
+                    .map(function(entry) {
+                        return entry[period];
+                    })
+                    .reduce(function(subtotalResult, subtotalValue) {
+                        return subtotalResult + subtotalValue;
+                    }, 0);
+                return periodResult;
+            }, {});
+
+            // Calculation for external ordering fee whether DITCO and OTHER
+
+            // TODO: fix the label below to respect DITCO/OTHER
+            var fees = {
+                "CLIN Type & Classification": "External ordering agency fee (" + payload.templatePayload.contractingShop.fee + "%)",
+            };
```

Comment:
One last thing to fix?  Need the expected output for both cases.

Reply:
Fixed in https://github.com/dod-ccpo/atat-snow/pull/347/commits/7e94f9e63318773e90a3205b498c08683b812e03

Followed naming [implemented here](https://github.com/dod-ccpo/atat-web-api/blob/89c9ceb57553e681cdd045b6dfefa77309f6fa1d/document-generation/igce-document.ts#L251)